feat(glue): Connection construct (#12444)
aws/aws-cdk: packages/@aws-cdk/aws-glue/README.md

This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.

## Connection

A `Connection` allows Glue jobs, crawlers and development endpoints to access certain types of data stores.

```ts
new glue.Connection(stack, 'MyConnection', {
  connectionType: glue.ConnectionTypes.NETWORK,
  securityGroups: [securityGroup],

>   subnet,

Reviewer: If we have in the example, lets just add a sentence or two into what these are - and why they are needed.

Author: Basing my comments on Subnet/Security groups section in https://docs.amazonaws.cn/en_us/glue/latest/dg/connection-defining.html

This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.

## Connection

A `Connection` allows Glue jobs, crawlers and development endpoints to access certain types of data stores.

```ts
new glue.Connection(stack, 'MyConnection', {
  connectionType: glue.ConnectionTypes.NETWORK,
  securityGroups: [securityGroup],
  subnet,
});
```

If you need to use a connection type that doesn't exist as a static member on `ConnectionType`, you can instantiate a `ConnectionType` object, e.g: `new glue.ConnectionType('NEW_TYPE')`.

> Different connection types require different connection properties which can be set via `ConnectionProps.properties`. 

Reviewer: No need.

Author: Do you mean line 40 or 38 as well? I think for 38 you mentioned it should be here - from the doc class docs?

Author: Dropped L40

> Subclasses of `Connection` are implemented to provide a connection type tailored props e.g. `MongoDBSourceConnection` which can be used as follows

Reviewer: ```suggestion
Concrete connection types are also available and provide a tailored, type-safe API for the specific connection options. 

### MongoDB
```

```ts
new glue.MongoDBSourceConnection(stack, 'Connection', {
    database: 'database',
    collection: 'collection',
});
```

> if a specific connection type construct props does not expose all properties, additional properties can still be set as follows

Reviewer: ```suggestion
In case a specific connection type construct does not expose all properties, additional properties can still be set using the `properties` property.
```
